tests/test_predicates.py

```diff
@@ -126,3 +127,90 @@ def test_disjoint():
+        ]
+    )
+
+    actual = spherely.touches(a, b)
+    expected = np.array([True, False])
+    np.testing.assert_array_equal(actual, expected)
+
+    a_p = spherely.point(1.0, 1.0)
+    b_p = spherely.point(1.0, 1.0)
+    # Points do not have a boundary, so they cannot touch per definition
+    # This is consistent with PostGIS for example (cmp. https://postgis.net/docs/ST_Touches.html)
+    assert not spherely.touches(a_p, b_p)
+
+    b_line = spherely.linestring([(1.0, 1.0), (1.0, 2.0)])
+    assert spherely.touches(a_p, b_line)
+
+
+@pytest.fixture
+def parent_poly():
+    return spherely.polygon(
+        [
+            (-118.0, 60.0),
+            (-118.0, 40.0),
+            (-118.0, 23.0),
+            (34.0, 23.0),
+            (34.0, 40.0),
+            (34.0, 60.0),
+        ]
+    )
+
+
+@pytest.fixture
+def polys_to_check_for_covers():
+    return (
+        np.array(
+            [
+                # Basic point covers tests, outside, on boundary and interior
+                spherely.point(-120.0, 70.0),
+                spherely.point(-118.0, 41.0),
+                spherely.point(-116.0, 37.0),
+                # Basic polyline tests, crossing, on boundary and interior
+                spherely.linestring([(-120.0, 70.0), (-116.0, 37.0)]),
+                spherely.linestring([(-118.0, 41.0), (-118.0, 23.0)]),
+                spherely.linestring([(-117.0, 39.0), (-115.0, 37.0)]),
+                # Basic polygon test, crossing, shared boundary and interior
+                spherely.polygon(
+                    [(-120.0, 41.0), (-120.0, 35.0), (-115.0, 35.0), (-115.0, 41.0)]
+                ),
+                # TODO: This case is currently not fully correct. Supplying a polygon `a` and `b` and checking
+                # whether `a` covers `b`, where `a` and `b` share co-linear edges only works when the edges
+                # between them are identical. An example of this breaking would be a polygon `a` consisting of the edges AB, BC and CA and
+                # a polygon `b` consisting of the edges AB*, B*C* and C*A, where B* and C* reside somewhere on the edge AB and BC, respectively.
+                # In this case, s2geometry tries to resolve the co-linearity by symbolic perturbation, where B* and C* are moved by an infinitesimal
+                # amount. However, the resulting geometry may then not be covered anymore, even if it would be in reality. Therefor, these tests assume
+                # identical shared edges between polygons `a` and `b`, which does work as intended.
+                spherely.polygon(
+                    [(-118.0, 40.0), (-118.0, 23.0), (34.0, 23.0), (34.0, 40.0)]
+                ),
+                spherely.polygon(
+                    [(-117.0, 40.0), (-117.0, 35.0), (-115.0, 35.0), (-115.0, 40.0)]
+                ),
+            ]
+        ),
+        np.array([False, True, True, False, True, True, False, True, True]),
```

Comment:
We don't really have extensive tests for `contains` with those corner case of on the boundary etc. Would it make sense to run this function for both, and so create a second numpy array with expected labels for `contains`? (that also makes it very explicit what the differences are, and whether we already implement the same nuance between contains vs covers or not)

Reply:
Yeah, definitely makes sense to have this! I have refactored it slightly, although the new test for `contains` fails with what I believe are the correct outputs.

Currently, `contains` returns true for a point on the boundary of a polygon, however, it should not as there is no point that is common between the interior of the polygon and the point in question (https://postgis.net/docs/ST_Contains.html)